src/python/pants/python/python_setup.py

```diff
@@ -43,16 +62,16 @@ def register_options(cls, register):
             type=file_option,
             help=(
                 "When resolving third-party requirements, use this "
-                "constraint file to determine which versions to use. See "
+                "constraints file to determine which versions to use. See "
                 "https://pip.pypa.io/en/stable/user_guide/#constraints-files for more information "
                 "on the format of constraint files and how constraints are applied in Pex and Pip."
             ),
         )
         register(
             "--resolve-all-constraints",
             advanced=True,
-            default=False,
-            type=bool,
+            default=ResolveAllConstraintsOption.NEVER,
```

Comment:
We may want to default to Non-Deployables. We'd need to update the `pex.py` code to not error if `--requirement-constraints` isn't set, though, which might be fine.

Reply:
I thought about that, wasn't sure if that made sense. Probably does though?

Reply:
I think it does. We really want people to use this option, and changing the default lowers the barrier to entry. Now all you need to do is set up a constraints.txt, which you should be doing anyways for reproducibility.

I also think we want people using the `nondependencies` option much more than we want them using `always`. If we set a default, we don't have as much risk of them incorrectly choosing the wrong thing.

Reply:
Done!